mongo/crud_examples_test.go

```diff
@@ -0,0 +1,525 @@
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("deleted %v documents\n", res.DeletedCount)
+}
+
+func ExampleCollection_Distinct() {
+	var coll *mongo.Collection
+
+	// find all unique values for the "name" field for documents in which the "age" field is greater than 25
+	// specify the MaxTime option to limit the amount of time the operation can run on the server
+	filter := bson.D{{"age", bson.D{{"$gt", 25}}}}
+	opts := options.Distinct().SetMaxTime(2 * time.Second)
+	values, err := coll.Distinct(context.TODO(), "name", filter, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, value := range values {
+		fmt.Println(value)
+	}
+}
+
+func ExampleCollection_EstimatedDocumentCount() {
+	var coll *mongo.Collection
+
+	// get and print an estimated of the number of documents in the collection
+	// specify the MaxTime option to limit the amount of time the operation can run on the server
+	opts := options.EstimatedDocumentCount().SetMaxTime(2 * time.Second)
+	count, err := coll.EstimatedDocumentCount(context.TODO(), opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("estimated document count: %v", count)
+}
+
+func ExampleCollection_Find() {
+	var coll *mongo.Collection
+
+	// find all documents in which the "name" field is "Bob"
+	// specify the Sort option to sort the returned documents by name in ascending order
+	opts := options.Find().SetSort(bson.D{{"name", 1}})
+	cursor, err := coll.Find(context.TODO(), bson.D{{"name", "Bob"}}, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// get a list of all returned documents and print them out
+	// see the mongo.Cursor documentation for more examples of using cursors
+	var results []bson.M
+	if err = cursor.All(context.TODO(), &results); err != nil {
+		log.Fatal(err)
+	}
+	for _, result := range results {
+		fmt.Println(result)
+	}
+}
+
+func ExampleCollection_FindOne() {
+	var coll *mongo.Collection
+
+	// find at most one document in which the "name" field is "Bob"
+	// specify the Sort option to sort the documents by age
+	// the first document in the sorted order will be returned
+	opts := options.FindOne().SetSort(bson.D{{"age", 1}})
+	var result bson.M
+	err := coll.FindOne(context.TODO(), bson.D{{"name", "Bob"}}, opts).Decode(&result)
+	if err != nil {
+		// ErrNoDocuments means that the filter did not match any documents in the collection
+		if err == mongo.ErrNoDocuments {
+			return
+		}
+		log.Fatal(err)
+	}
+	fmt.Printf("found document %v", result)
+}
+
+func ExampleCollection_FindOneAndDelete() {
+	var coll *mongo.Collection
+
+	// find and delete at most one document in which the "name" field is "Bob"
+	// specify the Sort option to sort the documents by age
+	// the first document in the sorted order will be deleted
+	opts := options.FindOneAndDelete().SetSort(bson.D{{"age", 1}})
+	var deletedDocument bson.M
+	err := coll.FindOneAndDelete(context.TODO(), bson.D{{"name", "Bob"}}, opts).Decode(&deletedDocument)
```

Comment:
Show the most common scenario: filtering by something unique.  For options, show projection instead of sort

Reply:
done.